src/Compilers/CSharp/Portable/Declarations/MergedTypeDeclaration.cs

```diff
@@ -123,6 +123,22 @@ public bool AnyMemberHasAttributes
             }
         }
 
+        public bool HasConstraints
+        {
+            get
+            {
+                foreach (var decl in this.Declarations)
```

Comment:
I followed the rest of the pattern here.  And i'm always wary of allocations in the compiler paths :)